lib/utils/github.ts

```diff
@@ -254,10 +253,16 @@ export async function openUsageIssue(
 /**
  * Get main branch of a repo
  */
-export const getMainBranch = async (repoFullName: string) => {
-	const repoRes = (await fetch(
-		`https://api.github.com/repos/${repoFullName}`
-	)) as Response
+export const getMainBranch = async (
+	repoFullName: string,
+	accessToken?: string
+) => {
+	const repoRes = (await fetch(`https://api.github.com/repos/${repoFullName}`, {
+		method: 'GET',
+		headers: {
+			Authorization: accessToken && `token ${accessToken}`
+		}
+	})) as Response
 	const repo = (await repoRes.json()) as any
```

Comment:
@sarimrmalik any idea how to make this response typesafe? Ie. by importing GH types